### ISSUE TITLE
default-prefix-filter is not availabe in ibm_tg_connection_prefix_filter

### ISSUE THREAD

**Issue by Ak-sky** [IBM-Cloud/terraform-provider-ibm#5261]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I noticed that `default-prefix-filter` is not available in [ibm_tg_connection_prefix_filter](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/tg_connection_prefix_filter#connection_id) resource, it is available in both [CLI](https://cloud.ibm.com/docs/transit-gateway?topic=transit-gateway-transit-gateway-cli&interface=ui#connection-create) and [API](https://cloud.ibm.com/apidocs/transit-gateway#create-transit-gateway-connection).

### New or Affected Resource(s)

* ibm_tg_connection_prefix_filter

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by Ak-sky** [comment]
@sushmitha1506

**Comment by anaghajoshiibm** [comment]
Thanks for raising this @Ak-sky 

default-prefix-filter is already supported in Terraform as default_prefix_filter, but it is part of the ibm_tg_connection resource (not ibm_tg_connection_prefix_filter).

The confusion is due to a documentation gap, as this field was not clearly documented earlier.

We have raised a PR to address this and improve the documentation:
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/6734

Once merged, the docs will clarify the correct usage and resource mapping.

**Comment by Ak-sky** [comment]
@anaghajoshiibm, I agree, In fact we have been using it here- https://github.com/terraform-ibm-modules/terraform-ibm-transit-gateway/blob/main/main.tf#L42-L52
It is just
- it was not documented.
- also why it is not part of `ibm_tg_connection_prefix_filter`, as it is confusing, can you please clarify.

**Comment by anaghajoshiibm** [comment]
Thanks for pointing this out @Ak-sky

I verified this from the API, SDK, and Terraform provider implementation to clarify the behaviour.

`default_prefix_filter` is a connection-level property (exposed as `PrefixFiltersDefault` in the API/SDK) and is configured as part of the `ibm_tg_connection` resource. It defines the fallback action applied when no prefix filter rules match.

On the other hand, `ibm_tg_connection_prefix_filter` is used to manage individual prefix filter rules (specific CIDR-based permit/deny entries). These are handled via separate API operations.

So the separation is intentional:

* `ibm_tg_connection_prefix_filter` → individual prefix rules
* `default_prefix_filter` (in `ibm_tg_connection`) → default/fallback behavior

I’ve updated the documentation to clarify this distinction and avoid confusion:
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/6734

**Comment by Ak-sky** [comment]
Thank you @anaghajoshiibm for the clarification, closing this issue.
